instrumentation/httptrace/config.go

```diff
@@ -53,6 +55,25 @@ func newConfig() config {
 		traceClientIP:                internal.BoolEnv(envTraceClientIPEnabled, false),
 		isStatusError:                isServerError,
 		inferredProxyServicesEnabled: internal.BoolEnv(envInferredProxyServicesEnabled, false),
+		baggageTagKeys:               make(map[string]struct{}),
+	}
+	raw, ok := os.LookupEnv("DD_TRACE_BAGGAGE_TAG_KEYS")
+	if !ok {
+		// not set → default list
+		raw = "user.id,account.id,session.id"
+	}
+	// Now raw=="” only if they explicitly set it to empty
+	parts := strings.Split(raw, ",")
+	for _, part := range parts {
+		key := strings.TrimSpace(part)
+		if key == "" {
+			continue
+		}
+		if key == "*" {
+			c.allowAllBaggage = true
+		} else {
+			c.baggageTagKeys[key] = struct{}{}
+		}
```

Comment:
What about something like this?
```suggestion
	if v, ok := os.LookupEnv("DD_TRACE_BAGGAGE_TAG_KEYS"); ok {
		if v == "*" {
			c.allowAllBaggage = true
		} else {
			for _, part := range strings.Split(v, ",") {
				key := strings.TrimSpace(part)
				if key == "" {
					continue
				} else {
					c.baggageTagKeys[key] = struct{}{}
				}
			}
		}
	} else {
		c.baggageTagKeys = defaultBaggageTagKeys()
	}
```

Where defaultBaggageTagKeys is:
```
func defaultBaggageTagKeys() map[string]struct{} {
	return map[string]struct{}{
		"user.id":    struct{}{},
		"account.id": struct{}{},
		"session.id": struct{}{},
	}
}
```

Reply:
Note that in my change:

- DD_TRACE_BAGGAGE_TAG_KEYS=* toggles allowAllBaggage to true
- DD_TRACE_BAGGAGE_TAG_KEYS=something,*,somethingelse does not toggle allowAllBaggage to true